internal/sleuth/incident_impact_source_resource.go

```diff
@@ -0,0 +1,876 @@
+
+	pd, diags := parseProviderData(ctx, plan)
+
+	tflog.Info(ctx, "Creating IncidentImpactSource resource", map[string]any{"name": plan.Name.ValueString(), "projectSlug": plan.ProjectSlug.ValueString()})
+
+	var pdInput *pagerDutyInputResourceModel
+	diags = plan.PagerDutyInput.As(ctx, &pdInput, basetypes.ObjectAsOptions{})
+	res.Diagnostics.Append(diags...)
+
+	tflog.Info(ctx, "Updating IncidentImpactSource resource", map[string]any{"plan": plan})
+
+	if res.Diagnostics.HasError() {
+		tflog.Error(ctx, "Error getting IncidentImpactSource plan", map[string]any{"diagnostics": res.Diagnostics})
+		return
+	}
+
+	projectSlug := plan.ProjectSlug.ValueString()
+	inputFields, diags := getMutableIncidentImpactSourceStruct(ctx, plan, pd)
+	res.Diagnostics.Append(diags...)
+	input := gqlclient.IncidentImpactSourceInputUpdateType{
+		Slug:                          state.Slug.ValueString(),
+		IncidentImpactSourceInputType: inputFields,
+	}
+
+	ccs, err := iisr.c.UpdateIncidentImpactSource(ctx, input)
+	if err != nil {
+		tflog.Error(ctx, "Error updating IncidentImpactSource", map[string]any{"error": err.Error()})
+		res.Diagnostics.AddError(
+			"Error updating IncidentImpactSource",
+			fmt.Sprintf("Could not update code change soure, unexpected error: %+v", err.Error()),
+		)
+		return
+	}
+
+	newState, diags := getNewStateFromIncidentImpactSource(ctx, ccs, projectSlug, pd)
+	res.Diagnostics.Append(diags...)
+
+	diags = res.State.Set(ctx, newState)
+	res.Diagnostics.Append(diags...)
+	tflog.Info(ctx, fmt.Sprintf("Successfully created IncidentImpactSource"), map[string]any{"diags": res.Diagnostics})
+
+}
+
+func (iisr *incidentImpactSourceResource) Delete(ctx context.Context, req resource.DeleteRequest, res *resource.DeleteResponse) {
+	ctx = tflog.SetField(ctx, "resource", "incident_impact_source")
+	ctx = tflog.SetField(ctx, "operation", "delete")
+
+	var state incidentImpactResourceModel
+	diags := req.State.Get(ctx, &state)
+	res.Diagnostics.Append(diags...)
+
+	tflog.Info(ctx, "Deleting IncidentImpactSource resource", map[string]any{"state": state})
+	projectSlug := state.ProjectSlug.ValueStringPointer()
+	slug := state.Slug.ValueStringPointer()
+
+	err := iisr.c.DeleteImpactSource(projectSlug, slug)
+	if err != nil {
+		tflog.Error(ctx, "Error deleting IncidentImpactSource", map[string]any{"error": err.Error()})
+		res.Diagnostics.AddError(
+			"Error deleting IncidentImpactSource",
+			fmt.Sprintf("Could not delete code change soure, unexpected error: %+v", err.Error()),
+		)
+		return
+	}
+}
+
+func (iisr *incidentImpactSourceResource) ImportState(ctx context.Context, req resource.ImportStateRequest, res *resource.ImportStateResponse) {
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, res)
+}
+
+func getNewStateFromIncidentImpactSource(ctx context.Context, iis *gqlclient.IncidentImpactSource, projectSlug string, data providerData) (incidentImpactResourceModel, diag.Diagnostics) {
+	iim := incidentImpactResourceModel{
```

Comment:
```suggestion
	iirm := incidentImpactResourceModel{
```
Or is it a convention to exclude "resource" from the variable abbreviation?